src/qs8-vadd/gen/qs8-vadd-minmax-hvx-u128.c

```diff
@@ -0,0 +1,182 @@
+    HVX_Vector vb1_mul_o = Q6_Vw_vmpyio_VwVh(vb_multiplier, vb1_i16_lo);
+    HVX_VectorPair va1_mul = Q6_W_vshuff_VVR(va1_mul_o, va1_mul_e, -4);
+    HVX_VectorPair vb1_mul = Q6_W_vshuff_VVR(vb1_mul_o, vb1_mul_e, -4);
+    HVX_Vector vacc1 = Q6_Vw_vadd_VwVw(vbias, Q6_V_lo_W(va1_mul));
+    vacc1 = Q6_Vw_vadd_VwVw(vacc1, Q6_V_lo_W(vb1_mul));
+    HVX_Vector va2_mul_e = Q6_Vw_vmpyio_VwVh(va_multiplier, Q6_Vh_vshuffe_VhVh(va2_i16_lo, va2_i16_lo));
+    HVX_Vector vb2_mul_e = Q6_Vw_vmpyio_VwVh(vb_multiplier, Q6_Vh_vshuffe_VhVh(vb2_i16_lo, vb2_i16_lo));
+    HVX_Vector va2_mul_o = Q6_Vw_vmpyio_VwVh(va_multiplier, va2_i16_lo);
+    HVX_Vector vb2_mul_o = Q6_Vw_vmpyio_VwVh(vb_multiplier, vb2_i16_lo);
+    HVX_VectorPair va2_mul = Q6_W_vshuff_VVR(va2_mul_o, va2_mul_e, -4);
+    HVX_VectorPair vb2_mul = Q6_W_vshuff_VVR(vb2_mul_o, vb2_mul_e, -4);
+    HVX_Vector vacc2 = Q6_Vw_vadd_VwVw(vbias, Q6_V_lo_W(va2_mul));
+    vacc2 = Q6_Vw_vadd_VwVw(vacc2, Q6_V_lo_W(vb2_mul));
+    HVX_Vector va3_mul_e = Q6_Vw_vmpyio_VwVh(va_multiplier, Q6_Vh_vshuffe_VhVh(va3_i16_lo, va3_i16_lo));
+    HVX_Vector vb3_mul_e = Q6_Vw_vmpyio_VwVh(vb_multiplier, Q6_Vh_vshuffe_VhVh(vb3_i16_lo, vb3_i16_lo));
+    HVX_Vector va3_mul_o = Q6_Vw_vmpyio_VwVh(va_multiplier, va3_i16_lo);
+    HVX_Vector vb3_mul_o = Q6_Vw_vmpyio_VwVh(vb_multiplier, vb3_i16_lo);
+    HVX_VectorPair va3_mul = Q6_W_vshuff_VVR(va3_mul_o, va3_mul_e, -4);
+    HVX_VectorPair vb3_mul = Q6_W_vshuff_VVR(vb3_mul_o, vb3_mul_e, -4);
+    HVX_Vector vacc3 = Q6_Vw_vadd_VwVw(vbias, Q6_V_lo_W(va3_mul));
+    vacc3 = Q6_Vw_vadd_VwVw(vacc3, Q6_V_lo_W(vb3_mul));
+
+    // right shift
+    vacc0 = Q6_Vw_vasr_VwR(vacc0, vshift);
+    vacc1 = Q6_Vw_vasr_VwR(vacc1, vshift);
+    vacc2 = Q6_Vw_vasr_VwR(vacc2, vshift);
+    vacc3 = Q6_Vw_vasr_VwR(vacc3, vshift);
+
+    // pack: 32 bit to 16 bit
+    HVX_Vector vout0 = Q6_Vh_vadd_VhVh(voutput_zero_point, Q6_Vh_vpack_VwVw_sat(vacc0, vacc0));
+    HVX_Vector vout1 = Q6_Vh_vadd_VhVh(voutput_zero_point, Q6_Vh_vpack_VwVw_sat(vacc1, vacc1));
+    HVX_Vector vout2 = Q6_Vh_vadd_VhVh(voutput_zero_point, Q6_Vh_vpack_VwVw_sat(vacc2, vacc2));
+    HVX_Vector vout3 = Q6_Vh_vadd_VhVh(voutput_zero_point, Q6_Vh_vpack_VwVw_sat(vacc3, vacc3));
+
+    // pack: 16 bit to 8 bit
+    vout0 = Q6_Vb_vpack_VhVh_sat(vout0, vout0);
+    vout1 = Q6_Vb_vpack_VhVh_sat(vout1, vout1);
+    vout2 = Q6_Vb_vpack_VhVh_sat(vout2, vout2);
+    vout3 = Q6_Vb_vpack_VhVh_sat(vout3, vout3);
+
+    // minmax
+    vout0 = Q6_Vb_vmax_VbVb(voutput_min, vout0);
+    vout0 = Q6_Vb_vmin_VbVb(voutput_max, vout0);
+    vout1 = Q6_Vb_vmax_VbVb(voutput_min, vout1);
+    vout1 = Q6_Vb_vmin_VbVb(voutput_max, vout1);
+    vout2 = Q6_Vb_vmax_VbVb(voutput_min, vout2);
+    vout2 = Q6_Vb_vmin_VbVb(voutput_max, vout2);
+    vout3 = Q6_Vb_vmax_VbVb(voutput_min, vout3);
+    vout3 = Q6_Vb_vmin_VbVb(voutput_max, vout3);
+
+    // store output
+    Q6_V_vstu_variable(ptr_o, 32, vout0); ptr_o += 32;
```

Comment:
Same comment about loading applies here too. We should be able to store one 128 byte vector here instead of 4 overlapping ones.

Reply:
Done!